### PR TITLE
Add xcode 7 and some fixes

### PR DESCRIPTION
**Pull request by imurchie**
Add iOS 7 simulator, and (currently empty) class for Xcode 7. 

@Jonahss @jlipps

### CODE REVIEW

File: lib/logger.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import { getLogger } from 'appium-logger';
-const log = getLogger('appium-ios-simulator');
+
+const log = getLogger('AppiumIosSimulator');
```
Comment (jlipps) [inline lib/logger.js]: for the sake of brevity, i'm fine calling this `IOSSim` or something

File: lib/simulator-xcode-6.js [hunk 2]
```diff
@@ -5,66 +5,119 @@ import { fs, util } from 'appium-support';
...
     //TODO assert that udid is a valid sim
-    this.keychainPath = path.resolve(this.getDir(), "Library", "Keychains");
+    this.keychainPath = path.resolve(this.getDir(), 'Library', 'Keychains');
+
+    this.appDataBundlePaths = null;
   }
 
   //TODO default constructor with no args should create a sim with some defaults
 
+  async getPlatformVersion () {
```
Comment (jlipps) [inline lib/simulator-xcode-6.js]: what about making this `async get platformVersion()`, then storing the actual value in an inaccessible cached version?

File: lib/simulator-xcode-6.js [hunk 3]
```diff
@@ -5,66 +5,119 @@ import { fs, util } from 'appium-support';
...
-  async getAppDataDir (bundleId) {
+  getLogDir () {
+    return path.resolve(this.getDir(), 'Library', 'Logs');
+  }
 
+  /*
+   * takes an `id`, which is either a bundleId (e.g., com.apple.mobilesafari)
+   * or, for iOS 7.1, the app name without `.app` (e.g., MobileSafari)
+   */
+  async getAppDataDir (id) {
```
Comment (jlipps) [inline lib/simulator-xcode-6.js]: same comment here about `async get`

File: lib/simulator-xcode-6.js [hunk 4]
```diff
@@ -103,19 +154,23 @@ class SimulatorXcode6 {
...
+    } else {
+      files.push('Applications');
+    }
 
     files = files.map((s) => {
       return path.resolve(this.getDir(), s);
     });
 
-    let existence = files.map((f) => {
+    let existence = files.map(async (f) => {
```
Comment (jlipps) [inline lib/simulator-xcode-6.js]: funny, i just added asyncMap to asyncbox, so instead you could do:

```js
let existence = await asyncMap(files.map, async (f) => { return util.hasAccess(f); });
```

it runs in parallel automatically.

up to you.

File: lib/simulator-xcode-6.js [hunk 5]
```diff
@@ -149,6 +204,26 @@ class SimulatorXcode6 {
...
+
+    // wait for the system to create the files we will manipulate
+    // need quite a high retry number, in order to accomodate iOS 7.1
+    // locally, 7.1 averages 8.5 retries (from 6 - 12)
+    //          8+ averages 0.6 retries (from 0 - 2)
+    const RETRIES = 15;
+    let retry;
+    for (retry = 0; retry < RETRIES; retry++) {
+      if (await this.isFresh()) {
+        if (retry < RETRIES-1) {
```
Comment (jlipps) [inline lib/simulator-xcode-6.js]: `RETRIES - 1`

**Comment by jlipps** [comment]
love the approach! and thanks for documenting everything in the README, lovely idea

File: lib/simulator-xcode-6.js (hunk 2 above)
Comment (imurchie) [inline lib/simulator-xcode-6.js]: I can't seem to get async getters to work at all. Babel (and JSHint) report a syntax error.

Comment (jlipps) [inline lib/simulator-xcode-6.js]: oh weird. maybe it's a thing that i invented in my es6-addled brain.

**Comment by Jonahss** [comment]
Cool

**Pull request merged**